pkg/auth/fleetshard_authz_middleware.go

```diff
@@ -47,3 +69,29 @@ func checkAllowedOrgIDs(allowedOrgIDs AllowedOrgIDs) mux.MiddlewareFunc {
 		})
 	}
 }
+
+func checkSubject(expected string) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			ctx := request.Context()
+			claims, err := GetClaimsFromContext(ctx)
+			if err != nil {
+				// Deliberately return 404 here so that it will appear as the endpoint doesn't exist if requests are
+				// not authorised. Otherwise, we would leak information about existing cluster IDs, since the path
+				// of the request is /agent-clusters/<id>.
+				shared.HandleError(request, writer, errors.NotFound(""))
+				return
+			}
+
+			s, _ := claims.GetSubject()
+			if expected == s {
+				next.ServeHTTP(writer, request)
+				return
+			}
+
+			glog.Infof("fleetshard subject is not allowed (expected=%s, actual=%s)", expected, s)
```

Comment:
In case the subject is empty or malformed:
```suggestion
			glog.Infof("fleetshard subject is not allowed (expected=%s, actual=%q)", expected, s)
```